lib/tasks_private/spec_helper.rake

```diff
@@ -4,15 +4,47 @@ namespace :spec do
     tower_host = ENV['TOWER_URL'] || "https://dev-ansible-tower3.example.com/api/v1/"
     id = ENV['TOWER_USER'] || 'testuser'
     password = ENV['TOWER_PASSWORD'] || 'secret'
-    PopulateTower.new(tower_host, id, password).create_dataset.counts
+    PopulateTower.new(tower_host, id, password, TowerData.new(true)).create_dataset.counts
   end
 
   desc "Get counts of various Tower objects"
   task :tower_counts do
     tower_host = ENV['TOWER_URL'] || "https://dev-ansible-tower3.example.com/api/v1/"
     id = ENV['TOWER_USER'] || 'testuser'
     password = ENV['TOWER_PASSWORD'] || 'secret'
-    PopulateTower.new(tower_host, id, password).counts
+    PopulateTower.new(tower_host, id, password, TowerData.new(false)).counts
+  end
+end
+
+class TowerData
+  def initialize(delete)
+    require ManageIQ::Providers::AnsibleTower::Engine.root.join('spec', 'support', 'tower_data_helper')
+
+    if delete || !Spec::Support::TowerDataHelper.tower_data_initialized?
+      init_blank
+    else
+      @data = Spec::Support::TowerDataHelper.tower_data
+      raise 'Invalid Tower data stored, can’t update.' unless @data.kind_of?(Hash)
+    end
+  end
+
+  def init_blank
+    @data = {}
+    write
+  end
+
+  def read
+    @data
+  end
+
+  def write
+    Spec::Support::TowerDataHelper.tower_data = @data
+  end
+
+  def update
+    yield @data if block_given?
+    write
```

Comment:
this write each time something updated seems too much.  yeah, it's not a big deal in this rake task. but ....

Reply:
I agree that this excessive writing just as the whole TowerData class might be an overkill. I’d just like to explain my motivation to that:

Because the population of a Tower instance can easily end with some failure during the process, I wanted to have the YAML file always consistent with what has been actually created in Tower. It’s always possible to revert it and it can serve as a reference.

But as @tumido pointed out, it is not actually necessary, because the YAML is intended to be used solely by Specs. And it can be used only if complete. For reference you can always read the terminal output or actually look into the Tower web UI.